### ISSUE TITLE
esbuild does not properly resolve extends field in tsconfig.json

### ISSUE THREAD

**Issue by haggholm** [evanw/esbuild#279]
If the `tsconfig.json` file contains an `extends` field, and this field is a package reference rather than pointing specifically to the JSON file, it is not properly resolved by esbuild.

For example, I may have a package `@my-scope/tsconfig-base`, where `package.json` specifies its `main` entry as `tsconfig.json`. Then this will work:

```json
{ "extends": "@my-scope/tsconfig-base/tsconfig.json" }
```
but this will not:
```json
{ "extends": "@my-scope/tsconfig-base" }
```
The Typescript documentation specifies that “The path may use Node.js style resolution”; I interpret this as meaning that it _should_ resolve `main` entries in referenced packages.

**Comment by evanw** [comment]
This doesn't appear to be how the official TypeScript compiler works. Here's what I did to test it:

* `entry.tsx`
    ```tsx
    console.log(<div/>)
    ```

* `tsconfig.json`
    ```json
    { "extends": "@my-scope/tsconfig-base" }
    ```

* `node_modules/@my-scope/tsconfig-base/package.json`
    ```json
    { "main": "./test.json" }
    ```

* `node_modules/@my-scope/tsconfig-base/test.json`
    ```json
    { "compilerOptions": { "jsx": "react", "jsxFactory": "worked" } }
    ```

Trying to build this with the official TypeScript compiler using `tsc --project .` gives the error `File '@my-scope/tsconfig-base' not found`. It appears to me that the official TypeScript compiler does not respect the `main` field of `package.json` when resolving the `extends` path of `tsconfig.json`.

**Comment by haggholm** [comment]
Embarrassing. Nonetheless, I have packages that build with the official TS compiler, but not with esbuild—I guess at this rate I don’t really know whether to regard this as being _by specification_ or as a side effect. My scoped config packages have their `main` entry set to `tsconfig.json`, and (hence) each has a `tsconfig.json` file in the package root directory: I guess the official compiler defaults to looking for the file by name rather than by looking at `package.json`.

**Comment by evanw** [comment]
This should be fixed in version 0.6.9.

**Comment by neves** [comment]
Now I'm having the opposite problem: 
https://gist.github.com/neves/05b17fce65cdc3c92010537a162b3c25 
`"extends": "@tsconfig/recommended"` works!
but
`"extends": "@tsconfig/recommended/tsconfig.json"` doesn't.
